flambe/metric/dev/perplexity.py

```diff
@@ -28,3 +30,42 @@ def compute(self, pred: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
         """
         entropy = self.entropy(pred, target).mean()
         return torch.exp(entropy)
+
+    def aggregate(self, state: dict, *args, **kwargs) -> Dict:
+        """Aggregates by only storing entropy per sample
+
+        Parameters
+        ----------
+        state: dict
+            the metric state
+        args: the pred, target tuple
+
+        Returns
+        -------
+        dict
+            the state dict
+        """
+        pred, target = args
+        if not state:
+            state['historic_score'] = []
+        state['historic_score'].append(self.entropy(pred, target).cpu().detach())
```

Comment:
I think we can be more efficient by keeping only the sum:
```python
logits = self.entropy(pred, target).cpu().detach()
state['score'] += logits.sum()
state['num_samples'] += len(logits)
```
Then in finalize:
```python
logits = self.entropy(pred, target).cpu().detach()
return torch.exp(state['scores'] / state['num_samples']).item()
```

Reply:
done